modules/internal/DefaultRectangular.chpl

```diff
@@ -1968,10 +1971,41 @@ module DefaultRectangular {
     const Adata = _ddata_shift(A.eltType, A.theData, Aidx);
     const Bidx = B.getDataIndex(Blo);
     const Bdata = _ddata_shift(B.eltType, B.theData, Bidx);
-    _simpleTransferHelper(A, B, Adata, Bdata, len);
+
+    type t = A.eltType;
+    const elemsizeInBytes = if isNumericType(t) then numBytes(t)
+                            else c_sizeof(t).safeCast(int);
+    const doParallelAssign = len:int*elemsizeInBytes >= parallelAssignThreshold;
+
+    if doParallelAssign {
+      _simpleParallelTransferHelper(A, B, Adata, Bdata, len);
+    }
+    else{
+      _simpleTransferHelper(A, B, Adata, Bdata, len);
+    }
+  }
+
+  private proc _simpleParallelTransferHelper(A, B, Adata, Bdata, len) {
+    const numTasks = if __primitive("task_get_serial") then
+                      1 else _computeNumChunks(len);
+    const lenPerTask = len:int/numTasks;
+
+    if debugDefaultDistBulkTransfer && numTasks > 1 then
+      chpl_debug_writeln("\tWill do parallel transfer with ", numTasks, " tasks");
+
+    coforall tid in 0..#numTasks {
+      const myOffset = tid*lenPerTask;
+
+      if tid == numTasks-1 {
+        _simpleTransferHelper(A, B, Adata, Bdata, len:int-myOffset, offset=myOffset);
+      }
+      else {
+        _simpleTransferHelper(A, B, Adata, Bdata, lenPerTask, offset=myOffset);
```

Comment:
Can we compute the actualLen and pass it in, instead of having the 2 calls? e.g. 

```chpl
const myLen = if tid == numTasks-1 then len:int-myOffset else lenPerTask;
_simpleTransferHelper(A, B, Adata, Bdata, myLen, myOffset);


```